harness/determined/common/experimental/resource_pool.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import workspace
+
+
+class ResourcePool:
+    """A class representing a resource pool object.
+
+    Attributes:
+        name: (str) The name of the resource pool.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        name: Optional[str] = None,
+    ):
+        """Create a resource pool object.
+
+        Arguments:
+            session: The session to use for API calls.
+            workspace_name: (Optional) Name of the resource pool.
+        """
+        self.name = name
+        self._session = session
+
+    @classmethod
+    def _from_bindings(
+        cls, resource_pool_bindings: bindings.v1ResourcePool, session: api.Session
+    ) -> "ResourcePool":
+        resource_pool = cls(session, name=resource_pool_bindings.name)
+        return resource_pool
+
+    def bind(
+        self,
+        workspace_names: List[str],
+    ) -> None:
+        """
+        Bind a resource pool to workspaces.
+
+        Arguments:
+            workspace_names (list(str)): The names of the workspaces to be bound.
+        """
+        req = bindings.v1BindRPToWorkspaceRequest(
+            resourcePoolName=self.name,
+            workspaceNames=workspace_names,
+        )
+
+        bindings.post_BindRPToWorkspace(self._session, body=req, resourcePoolName=self.name)
+
+    def unbind(
+        self,
+        workspace_names: List[str],
+    ) -> None:
+        """
```

Comment:
A description in the docstring about what happens in edge cases (like if the workspaces don't resolve to names, if they're already unbound, etc.) would be helpful to a caller.

Reply:
`bind` and `set` have fewer edge cases, but I think still warrant a "what happens when the name isn't found" description.